src/librustc_passes/liveness.rs

```diff
@@ -903,30 +907,92 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
         self.rwu_table.assign_unpacked(idx, rwu);
     }
 
-    fn compute(&mut self, body: &hir::Expr<'_>) -> LiveNode {
-        debug!("compute: using id for body, {:?}", body);
+    fn compute(
+        &mut self,
+        fk: FnKind<'_>,
+        body: &hir::Body<'_>,
+        span: Span,
+        id: hir::HirId,
+    ) -> LiveNode {
+        debug!("compute: using id for body, {:?}", body.value);
+
+        let mut only_by_ref_upvars = true;
```

Comment:
Comment here would be nice -- 

* why do we care if we only have by-ref upvars?
* what is the role of the `self.acc` below (presumably so that we assume that by-ref upvars are used after the closure completes)